### PR TITLE
Reorganizing tutorial to include an Intermediate version

### PR DESCRIPTION
**Pull request by kab163**
There will be an Intro Tutorial (the previous tutorial used for ECP)
and an Intermediate Tutorial (the fractal-based tutorial for HPCIC and RADIUSS AWS tutorial series)
in one repo.
The CMake changes should be good to go!
On lassen with CUDA, I build with:
```
module load cmake/3.20.2
module load gcc/8.3.1
module load cuda/10.2.89
cmake -DENABLE_CUDA=On -DBLT_CXX_STD=c++14 -DCMAKE_CUDA_ARCHITECTURES=70 -DCMAKE_CUDA_FLAGS="--extended-lambda" ../
```
For host-only build:
```
cmake -DENABLE_CUDA=Off -DBLT_CXX_STD=c++14 ../
```
For HIP build: 
```
module load rocm/5.6.0
module load cmake/3.23.1
cmake -DROCM_ROOT_DIR=/opt/rocm-5.6.0 -DHIP_ROOT_DIR=/opt/rocm-5.6.0/hip -DHIP_PATH=/opt/rocm-5.6.0/llvm/bin -DBLT_CXX_STD=c++14 -DENABLE_HIP=On -DENABLE_OPENMP=Off -DENABLE_CUDA=Off -DCMAKE_CXX_COMPILER=/opt/rocm-5.6.0/llvm/bin/amdclang++ -DCMAKE_C_COMPILER=/opt/rocm-5.6.0/llvm/bin/amdclang -DCMAKE_HIP_ARCHITECTURES=gfx90a ../
```

TODO:

- [x] Ensure this builds on host
- [x] Ensure this builds on AMD platforms

### CODE REVIEW

File: Intermediate_Tutorial/CMakeLists.txt [hunk 1]
```diff
@@ -9,11 +9,12 @@ add_subdirectory(01-LOOP)
 
 if(RAJA_ENABLE_CUDA)
   add_subdirectory(02-CUDA)
+  add_subdirectory(04-TEAMS)
```
Comment (artv3) [inline Intermediate_Tutorial/CMakeLists.txt:12]: @kab163  we ended up calling the API "Launch"

**Comment by kab163** [comment]
I am getting an error in the Dockerfile:
```
unauthorized: unauthenticated: User cannot be authenticated with the token provided.
Error: Process completed with exit code 1.
```
But I think it is either the need for the `BLT_CXX_STD=c++14` flag or an update to the version of g++ it's using in the docker container.. @artv3 @davidbeckingsale any ideas?

**Comment by artv3** [comment]
> I am getting an error in the Dockerfile:
> 
> ```
> unauthorized: unauthenticated: User cannot be authenticated with the token provided.
> Error: Process completed with exit code 1.
> ```
> 
> But I think it is either the need for the `BLT_CXX_STD=c++14` flag or an update to the version of g++ it's using in the docker container.. @artv3 @davidbeckingsale any ideas?

Would it help to update the standard directly in cmake? -DCMAKE_CXX_STANDARD=14 ?

File: Intermediate_Tutorial/01-LOOP/fractal-ex1-RAJA-loop.cpp [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  width = atoi(argv[1]);
+  if (width < 10) {fprintf(stderr, "edge_length must be at least 10\n"); exit(-1);}
+
+  dx = (xMax - xMin) / width;
+  dy = (yMax - yMin) / width;
+
+  printf("computing %d by %d fractal with a maximum depth of %d\n", width, width, maxdepth);
+
+  //TODO: Create an Umpire QuickPool allocator with Unified Memory that will hold the
+  // pixels of the fractal image.
```
Comment (artv3) [inline Intermediate_Tutorial/01-LOOP/fractal-ex1-RAJA-loop.cpp:33]:
```suggestion
  //pixels of the fractal image.
```

**Pull request merged**